### PR TITLE
Added unit tests for replace functions

### PR DESCRIPTION
**Pull request by bonniecpk**
- replaceCloudCredentials()
- replaceDevToken()
- replaceRefreshToken()

### CODE REVIEW

File: oauthdoctor/oauth/helper.go [hunk 1]
```diff
@@ -169,59 +173,55 @@ func (c *Config) diagnose(err error) {
 	}
 }
 
+var (
```
Comment (bobhancock) [inline oauthdoctor/oauth/helper.go]: Nice use of a function literal.

File: oauthdoctor/oauth/helper.go [hunk 2]
```diff
@@ -169,59 +173,55 @@ func (c *Config) diagnose(err error) {
...
+	log.Print("Follow this guide to setup your OAuth2 client ID and client secret: " +
 		"https://developers.google.com/adwords/api/docs/guides/first-api-call#set_up_oauth2_authentication")
-	fmt.Print("New Client ID >> ")
-
-	reader := bufio.NewReader(os.Stdin)
-	input, _ := reader.ReadString('\n')
 
-	strings.Replace(input, "\n", "", -1)
-	clientID := strings.Replace(input, "\n", "", -1)
+	clientID := readClientID()
```
Comment (bobhancock) [inline oauthdoctor/oauth/helper.go]: Would get be a better verb than read in this case?

File: oauthdoctor/oauth/helper.go [hunk 3]
```diff
@@ -169,59 +173,55 @@ func (c *Config) diagnose(err error) {
...
+func replaceRefreshToken(c ConfigWriter, refreshToken string) {
 	log.Print("Would you like to replace your refresh token in the " +
 		"client library config file with the new one generated?")
-	fmt.Print("Enter Y for Yes [Anything else is No] >> ")
 
-	reader := bufio.NewReader(os.Stdin)
-	answer, _ := reader.ReadString('\n')
-
-	answer = strings.Replace(answer, "\n", "", -1)
+	fmt.Print("Enter Y for Yes [Anything else is No] >> ")
```
Comment (bobhancock) [inline oauthdoctor/oauth/helper.go]: By anything else, does that mean a blank? i.e., you press return without entering a value?

File: oauthdoctor/oauth/helper_test.go [hunk 4]
```diff
@@ -3,9 +3,163 @@ package oauth
...
+
+	replaceCloudCredentials(&test.c)
+
+	if test.c.cfgFile.ClientID != test.clientID || test.c.cfgFile.ClientSecret != test.clientSecret {
+		t.Errorf("[%s] got: (ClientID=%s, ClientSecret=%s), want: (ClientID=%s, ClientSecret=%s)",
+			test.desc, test.c.cfgFile.ClientID, test.c.cfgFile.ClientSecret, test.clientID, test.clientSecret)
+	}
+}
+
+func setup() func() {
```
Comment (bobhancock) [inline oauthdoctor/oauth/helper_test.go]: Consider renaming this function to be more explicit about the setup that is being done. setup() is usually the name of a procedure called in TestMain().

File: oauthdoctor/oauth/helper_test.go [hunk 5]
```diff
@@ -3,9 +3,163 @@ package oauth
...
+	}
+
+	for _, test := range tests {
+		readStdin = func() string {
+			return test.stdin
+		}
+
+		replaceRefreshToken(&test.c, test.input)
+
+		if test.want != test.c.cfgFile.RefreshToken {
```
Comment (bobhancock) [inline oauthdoctor/oauth/helper_test.go]: Reverse the order of variables in this test. It should always be got - want.

File: oauthdoctor/oauth/helper_test.go [hunk 6]
```diff
@@ -3,9 +3,163 @@ package oauth
...
+	readClientID = func() string {
+		return test.clientID
+	}
+	readClientSecret = func() string {
+		return test.clientSecret
+	}
+
+	replaceCloudCredentials(&test.c)
+
+	if test.c.cfgFile.ClientID != test.clientID || test.c.cfgFile.ClientSecret != test.clientSecret {
```
Comment (bobhancock) [inline oauthdoctor/oauth/helper_test.go]: got - want

File: oauthdoctor/oauth/helper.go (hunk 3 above)
Comment (bonniecpk) [reply oauthdoctor/oauth/helper.go]: A "enter" key is considered a No. So you are correct.

File: oauthdoctor/oauth/helper.go (hunk 1 above)
Comment (bonniecpk) [reply oauthdoctor/oauth/helper.go]: Thanks!

File: oauthdoctor/oauth/helper_test.go (hunk 6 above)
Comment (bonniecpk) [reply oauthdoctor/oauth/helper_test.go]: I am not entirely sure how you want me to change this. `replaceCloudCredentials(&test.c)` is changing the config values in-place, so it does not return the changed value (i.e. got).

Comment (bobhancock) [reply oauthdoctor/oauth/helper_test.go]: sgtm

**Pull request merged**